File: .sdk-repos/orchestration-cluster-api-csharp/examples/AgentDefinition.cs
// Compilable usage examples for agent definition operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class AgentDefinitionExamples
{
    #region GetAgentDefinition
    // <GetAgentDefinition>
    public static async Task GetAgentDefinitionExample(AgentDefinitionKey agentDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetAgentDefinitionAsync(agentDefinitionKey);
        Console.WriteLine($"Agent definition: {result.AgentDefinitionKey}, name: {result.Name}, type: {result.AgentType}");
    }
    // </GetAgentDefinition>
    #endregion GetAgentDefinition

    #region SearchAgentDefinitions

    // <SearchAgentDefinitions>
    public static async Task SearchAgentDefinitionsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchAgentDefinitionsAsync(new AgentDefinitionSearchQuery
        {
            Filter = new AgentDefinitionFilter
            {
                AgentType = AgentDefinitionTypeEnum.AIAGENTTASK,
            },
            Page = new LimitPagination
            {
                Limit = 50,
            },
        });

        foreach (var def in result.Items)
        {
            Console.WriteLine($"Agent definition: {def.AgentDefinitionKey}, name: {def.Name}, type: {def.AgentType}");
        }
    }
    // </SearchAgentDefinitions>
    #endregion SearchAgentDefinitions
}

File: .sdk-repos/orchestration-cluster-api-python/examples/agent_definition.py
# Compilable usage examples for agent definition operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    AgentDefinitionKey,
    AgentDefinitionSearchQuery,
    CamundaClient,
)


# region GetAgentDefinition
def get_agent_definition_example(agent_definition_key: AgentDefinitionKey) -> None:
    client = CamundaClient()

    agent_definition = client.get_agent_definition(
        agent_definition_key=agent_definition_key
    )

    print(f"Agent definition name: {agent_definition.name}")


# endregion GetAgentDefinition


# region SearchAgentDefinitions
def search_agent_definitions_example() -> None:
    client = CamundaClient()

    result = client.search_agent_definitions(data=AgentDefinitionSearchQuery())

    for agent_definition in result.items:
        print(f"Agent definition key: {agent_definition.agent_definition_key}")


# endregion SearchAgentDefinitions
